Modules/_dbmmodule.c

```diff
@@ -470,10 +470,22 @@ dbmopen_impl(PyObject *module, PyObject *filename, const char *flags,
         return NULL;
     }
 
-    PyObject *filenamebytes = PyUnicode_EncodeFSDefault(filename);
-    if (filenamebytes == NULL) {
+    PyObject *filenamecharsorbytes = PyOS_FSPath(filename);
```

Comment:
Use `PyUnicode_FSConverter()`.

Reply:
Please review again, I'm not sure of `PyUnicode_FSConverter` usage.